src/main/g8/test/forms/behaviours/FormBehaviours.scala

```diff
@@ -55,6 +55,34 @@ trait FormBehaviours extends FormSpec {
       checkForError(form, data, expectedError)
     }
   }
+  
+  def formWithMaxLengthTextFields(fields: MaxLengthField*) = {
+    for (field <- fields) {
+      s"fail to bind when ${field.fieldName} has more characters than ${field.maxLength}" in {
+        val invalid = "A" * (field.maxLength + 1)
+        val validFields = validData - field.fieldName
+        val data = validFields ++ Map(field.fieldName -> invalid)
+        val expectedError = error(field.fieldName, field.errorMessageKey, field.maxLength)
+        checkForError(form, data, expectedError)
+      }
+    }
+  }
+
+  def formWithMandatoryTextFieldsAndCustomKey(fields: (String, String)*) = {
```

Comment:
This is very similar to `formWithMandatoryTextFields`. Could we simply improve the existing function rather than having a new one? Instead of taking a varargs of tuples, it could take a varargs of a case class which lets you specify the field and the error key, defaulting the error key to "error.required".

This would mean updating the scaffolds too, but would be cleaner than a proliferation of very similar methods.

Reply:
:+1: